encord/orm/label_row.py

```diff
@@ -209,6 +223,23 @@ class LabelStatus(Enum):
     REVIEWED_TWICE = "REVIEWED_TWICE"
 
 
+@dataclass(frozen=True)
+class LabelRowExportHistory:
+    export_hash: str
+    key_hash: Optional[str]
+    user_email: Optional[str]
```

Comment:
I'm personally not a fan of these optional values. What if next to the `key_hash` there would be an additional value now. The user would need to understand the relationship between the `key_hash` the other thing, and the `user_email`. 

Can we have this baked into the type system. E.g.

```py
@dataclass(frozen=True)
class LabelRowExportHistoryApiKeyItem:
    export_hash: str
    exported_at: str
    key_hash: str
    
@dataclass(frozen=True)
class LabelRowExportHistoryUserItem:
    export_hash: str
    exported_at: str
    user_email: str
    
LabelRowExportHistory = Union[LabelRowExportHistoryApiKeyItem, LabelRowExportHistoryUserItem]

```

Reply:
An alternative would be something like that:

```py
@dataclass(frozen=True)
class LabelRowExportHistoryItem:
    export_hash: str
    exported_at: str
    identifier_type: KeyTypeEnum   # optionally this as a helper, but actually not needed.
    identifier_values: Union[ApiKeyIdentifier, UserIdentifier]
  
```
The first thing is my preferred solution though, because the payload type tells you exactly what you will get and what you should expect.